### PR TITLE
CI: Generate `CRATES` using shell

### PR DESCRIPTION
**Pull request by tcharding**
Instead of manually setting the crates list it is less error prone to do so mechanically. This required some changes to the `run_task` script which have now merged, so we update to use the new commit hash at the same time.
    
- Use shell to set the `CRATES` env var used by `run_task.sh`.
- Use latest revision of  rust-bitcoin-maintainer-tools

### CODE REVIEW

**Comment by apoelstra** [comment]
LGTM. What is holding up this coming out of draft?

I guess, we may want to encapsulate the `cargo metadata` thing into a function call with a nice name, but given how small the total script is, it might not be worth it.

**Comment by tcharding** [comment]
We need https://github.com/rust-bitcoin/rust-bitcoin-maintainer-tools/pull/13 to merge

**Comment by tcharding** [comment]
Adding an `echo $CRATES` statment and running `bash contrib/crates.sh` gives:

```bash
addresses base58 bitcoin fuzz hashes internals io primitives units
```

**Review by Kixunil — APPROVED** [review]
ACK 90e073ef3ea25ad0cea68d7efe8b7ad997fdf5f8

**Review by apoelstra — APPROVED** [review]
ACK 90e073ef3ea25ad0cea68d7efe8b7ad997fdf5f8 successfully ran local tests

**Pull request merged**
